docs/component-guides/profiles-irsa.md

```diff
@@ -0,0 +1,125 @@
+
+Below are the steps to configure IRSA to be used with Kubeflow Profiles.
+
+# Configuration Steps
+
+1. Export the following variables for convenience
+    ```
+    export CLUSTER_NAME=<your cluster name>
+    export REGION=<your region>
+    export AWS_ACCOUNT_ID=<your aws account id>
+    export PROFILE_NAME=<the name of the profile to be created>
+    ```
+
+1. Create an IAM policy using [the following policy document](../../awsconfigs/infra_configs/iam_profile_controller_policy.json)
+    ```
+    aws iam create-policy \
+    --region $REGION \
+    --policy-name kf-profile-controller-policy \
+    --policy-document file://awsconfigs/infra_configs/iam_profile_controller_policy.json
+    ```
+
+1. Create an IRSA for the profile controller using the policy
+    ```
+    eksctl create iamserviceaccount \
+    --cluster=$CLUSTER_NAME \
+    --name="profiles-controller-service-account" \
+    --namespace=kubeflow \
+    --attach-policy-arn="arn:aws:iam::${AWS_ACCOUNT_ID}:policy/kf-profile-controller-policy" \
+    --region=$REGION \
+    --override-existing-serviceaccounts \
+    --approve
+    ```
+
+1. Associate IAM OIDC with the cluster
+    ```
+    aws --region $REGION eks update-kubeconfig --name $CLUSTER_NAME
+
+    eksctl utils associate-iam-oidc-provider --cluster $CLUSTER_NAME --region $REGION --approve
+    ```
+
+1. Create an IAM trust policy to authorize federated requests from the OIDC provider
+    ```
+    export OIDC_URL=$(aws eks describe-cluster --region $REGION --name $CLUSTER_NAME  --query "cluster.identity.oidc.issuer" --output text | cut -c9-)
+
+    cat <<EOF > trust.json
+    {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+        "Effect": "Allow",
+        "Principal": {
+            "Federated": "arn:aws:iam::${AWS_ACC_NUM}:oidc-provider/${OIDC_URL}"
+        },
+        "Action": "sts:AssumeRoleWithWebIdentity",
+        "Condition": {
+            "StringEquals": {
+            "${OIDC_URL}:aud": "sts.amazonaws.com"
+            }
+        }
+        }
+    ]
+    }
+    EOF
+    ```
+
+1. Create an IAM policy to scope the permissions for the profile. For simplicity, we will use the `arn:aws:iam::aws:policy/AmazonS3FullAccess` policy as an example.
+
+1. Create the IAM role to be used for the profile IRSA
+    ```
+    aws iam create-role --role-name $PROFILE_NAME-$CLUSTER_NAME-role --assume-role-policy-document file://trust.json
+
+    aws iam attach-role-policy --role-name $PROFILE_NAME-$CLUSTER_NAME-role --policy-arn arn:aws:iam::aws:policy/AmazonS3FullAccess
+    ```
+
+1. Get the OIDC role arn for the created profile
+    ```
+    export OIDC_ROLE_ARN=$(aws iam get-role --role-name $PROFILE_NAME-$CLUSTER_NAME-role --output text --query 'Role.Arn')
+    ```
+
+1. Configure the profile user credentials via your auth provider
+   - If using `Dex` add the following user to the base deployment config map:
```

Comment:
again which config map. 

Also updating the config map doesnt take effect dynamically, one has to restart the relevant pods. 
Can we create a separate namespace and improve the experience here?

Reply:
removed this step as it is not necessary to test profile-irsa

it is sufficient to create a new profile within an existing user